tests/unit_tests/test_oedisi_tools/test_multi_container.py

```diff
@@ -0,0 +1,91 @@
+              
+    result = runner.invoke(cli, ['build', "-m"])
+    assert result.exit_code == 0
+    
+@pytest.mark.usefixtures('test_mc_build')
+def test_api_heath_endpoint(base_path: Path, monkeypatch: pytest.MonkeyPatch):
+    build_path = base_path / "build"
+    assert build_path.exists(), "Build path for the test project does not exist."
+    for folder in build_path.iterdir():
+        if folder.is_dir() and folder.name not in ["kubernetes", 'tester']:
+            assert (folder/"server.py").exists(), f"Server.py does not exist for path {folder}"
+            sys.path.insert(1, str(folder.absolute()))
+            module = importlib.import_module('server') 
+            app = getattr(module, "app")  
+            client = TestClient(app)
+            response = client.get("/")
+            assert response.status_code == 200
+            HeathCheck.validate(response.json())
+            sys.path.remove(str(folder.absolute()))
+
+@pytest.mark.skipif(IN_GITHUB_ACTIONS,  reason="test runs locally but fails on github actions")
+@pytest.mark.usefixtures('test_mc_build')
+def test_api_run(base_path: Path, monkeypatch: pytest.MonkeyPatch):
+    build_path = base_path / "build"
+    assert build_path.exists(), "Build path for the test project does not exist."
+    clients = {}
+    for folder in build_path.iterdir():
+        if folder.is_dir() and folder.name not in ["kubernetes", 'tester']:
+            assert (folder/"server.py").exists(), f"Server.py does not exist for path {folder}"
+            sys.path.insert(1, str(folder.absolute()))
+            module = importlib.import_module('server') 
+            app = getattr(module, "app")  
+            client = TestClient(app)
+            clients[folder.name] = client
+
+    assert BROKER_SERVICE in clients, f"No broker client in list of tested services. Available services {list(clients.keys())}"
+    client = clients[BROKER_SERVICE]
+    response = client.post("/run")
+    assert response.status_code == 200
+        
+@pytest.mark.skipif(IN_GITHUB_ACTIONS,  reason="Requires docker deamon running on the machine. Will not work with github actions")       
+@pytest.mark.usefixtures('test_mc_build')
+def test_docker_compose(base_path: Path, monkeypatch: pytest.MonkeyPatch):
+    build_path = base_path / "build"
+    assert build_path.exists(), "Build path for the test project does not exist."
+    monkeypatch.chdir(build_path)
+
+    import subprocess
+    import time
+    with subprocess.Popen(["docker-compose", "up", "-d"], stdout=subprocess.PIPE) as proc:
+        stdout = b""
+        while True:
+            stdout += proc.stdout.read()
+            output = stdout.decode('utf8')
+            assert "error" not in output.lower(), "Error running the docker compose file."
+            count = output.count("Started") 
+            if count >= 3:
+                break
+            time.sleep(0.1)
+    
+    subprocess.Popen(["docker-compose", "down"]) 
```

Comment:
This last one could be an integration test, and it might work properly there. I know there's other people using github actions to run docker compose.